### PR TITLE
z_bg_gnd_soulmeiro.c OK

### PR DESCRIPTION
**Pull request by i82orbom**
func_8087AF38 has a lot of temps, suggestions are welcome for naming.

### CODE REVIEW

File: src/overlays/actors/ovl_Bg_Gnd_Soulmeiro/z_bg_gnd_soulmeiro.c [hunk 1]
```diff
@@ -31,17 +31,182 @@ const ActorInit Bg_Gnd_Soulmeiro_InitVars = {
...
+};
+
+static InitChainEntry sInitChain[] = {
+    ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneForward, 1000, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneScale, 1000, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneDownward, 1000, ICHAIN_STOP),
+};
+
+static Vec3f zeroVec = { 0.0f, 0.0f, 0.0f };
```
Comment (Roman971) [inline src/overlays/actors/ovl_Bg_Gnd_Soulmeiro/z_bg_gnd_soulmeiro.c:48]:
```suggestion
static Vec3f sZeroVec = { 0.0f, 0.0f, 0.0f };
```

File: src/overlays/actors/ovl_Bg_Gnd_Soulmeiro/z_bg_gnd_soulmeiro.c [hunk 2]
```diff
@@ -31,17 +31,182 @@ const ActorInit Bg_Gnd_Soulmeiro_InitVars = {
...
+    Vec3f vecB;
+    s16 temp_1;
+    s16 temp_2;
+    f32 temp_3;
+    f32 temp_4;
+    f32 distXZ;
+    s32 i;
+    BgGndSoulmeiro* this2;
+
+    (this->unk_198) == 0 ? 0 : ((this->unk_198--));
```
Comment (Roman971) [inline src/overlays/actors/ovl_Bg_Gnd_Soulmeiro/z_bg_gnd_soulmeiro.c:107]: Can this be written with a simple if?
```suggestion
    if (this->unk_198 != 0) {
        this->unk_198--;
    }
```

File: src/overlays/actors/ovl_Bg_Gnd_Soulmeiro/z_bg_gnd_soulmeiro.c [hunk 3]
```diff
@@ -31,17 +31,182 @@ const ActorInit Bg_Gnd_Soulmeiro_InitVars = {
...
+
+    if (this->unk_198 == 20) {
+        Flags_SetSwitch(globalCtx, (this->actor.params >> 8) & 0x3F);
+        this->actor.draw = NULL;
+    }
+
+    // This should be this->unk_198 != 0, this is required to match
+    if (!this->unk_198) {
```
Comment (Roman971) [inline src/overlays/actors/ovl_Bg_Gnd_Soulmeiro/z_bg_gnd_soulmeiro.c:115]: Does `== 0` not work? (`!= 0` isn't equivalent to `!this->unk_198`)
```suggestion
    if (this->unk_198 == 0) {
```

File: src/overlays/actors/ovl_Bg_Gnd_Soulmeiro/z_bg_gnd_soulmeiro.c [hunk 4]
```diff
@@ -31,17 +31,182 @@ const ActorInit Bg_Gnd_Soulmeiro_InitVars = {
...
+        }
+    }
+}
+
+void func_8087B284(BgGndSoulmeiro* this, GlobalContext* globalCtx) {
+    s32 pad;
+
+    if (!Flags_GetSwitch(globalCtx, (this->actor.params >> 8) & 0x3F)) {
+        this->actor.draw = BgGndSoulmeiro_Draw;
+        if ((this->collider.base.acFlags & 2) != 0) {
```
Comment (Roman971) [inline src/overlays/actors/ovl_Bg_Gnd_Soulmeiro/z_bg_gnd_soulmeiro.c:161]:
```suggestion
        if (this->collider.base.acFlags & 2) {
```

File: src/overlays/actors/ovl_Bg_Gnd_Soulmeiro/z_bg_gnd_soulmeiro.c (hunk 2 above)
Comment (petrie911) [reply src/overlays/actors/ovl_Bg_Gnd_Soulmeiro/z_bg_gnd_soulmeiro.c:107]: For that matter, can it be `DECR(this->unk_198);`?

Comment (Roman971) [reply src/overlays/actors/ovl_Bg_Gnd_Soulmeiro/z_bg_gnd_soulmeiro.c:107]: Probably, but we prefer not to do that

Comment (i82orbom) [reply src/overlays/actors/ovl_Bg_Gnd_Soulmeiro/z_bg_gnd_soulmeiro.c:107]: DECR won't actually work, causing some regalloc issues

File: src/overlays/actors/ovl_Bg_Gnd_Soulmeiro/z_bg_gnd_soulmeiro.c (hunk 3 above)
Comment (i82orbom) [reply src/overlays/actors/ovl_Bg_Gnd_Soulmeiro/z_bg_gnd_soulmeiro.c:115]: Unfortunately not, if I remember from my past days when learning very old C, there was a tendency to convert this like var == 0 to !var, even though they did not represent a boolean. In this case the former wouldn't work.

Comment (Roman971) [reply src/overlays/actors/ovl_Bg_Gnd_Soulmeiro/z_bg_gnd_soulmeiro.c:117]: In that case I think the comment should be updated at least, so it's not confusing

**Review by Roman971 — APPROVED** [review]

**Review by fig02 — APPROVED** [review]

**Pull request merged**